fcrepo-transform/src/main/java/org/fcrepo/transform/TransformationFactory.java

```diff
@@ -56,7 +59,7 @@ public TransformationFactory() {
     public <T> Transformation<T> getTransform(final MediaType contentType, final InputStream inputStream) {
         final String mimeType = contentType.toString();
         if (mimeToTransform.containsKey(mimeType)) {
-            return (Transformation<T>) mimeToTransform.get(contentType.toString()).newTransform(inputStream);
+            return mimeToTransform.get(contentType.toString()).apply(inputStream);
```

Comment:
It's currying. The whole type is: `String` -> `InputStream` -> `RdfStream` ->  `Something` (whatever the `Transformation` goes to).